crates/cairo-lint-core/tests/test_files/double_parens/double_parens

```diff
@@ -0,0 +1,234 @@
+//! > assert expressions
+
+//! > cairo_code
+fn main() {
+    assert_eq!(((4)), 4);
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: unnecessary double parentheses found. Consider removing them.
+ --> lib.cairo:2:16
+    assert_eq!(((4)), 4);
+               ^***^
+
+//! > fixed
+fn main() {
+    assert_eq!(4, 4);
+}
+
+//! > ==========================================================================
+
+//! > double parens in let statement
+
+//! > cairo_code
+fn main() {
+    let x = ((10 * 2));
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: unnecessary double parentheses found. Consider removing them.
+ --> lib.cairo:2:13
+    let x = ((10 * 2));
+            ^********^
+
+warning[E0001]: Unused variable. Consider ignoring by prefixing with `_`.
+ --> lib.cairo:2:9
+    let x = ((10 * 2));
+        ^
+
+//! > fixed
+fn main() {
+    let _x = 10 * 2;
+}
+
+//! > ==========================================================================
+
+//! > double parens in match arm
+
+//! > cairo_code
+fn main() -> felt252 {
+    let x = 5;
+    match x {
+        1 => ((10)),
+        5 => ((20)),
+        _ => ((30)),
+    }
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: unnecessary double parentheses found. Consider removing them.
+ --> lib.cairo:4:14
+        1 => ((10)),
+             ^****^
+
+warning: Plugin diagnostic: unnecessary double parentheses found. Consider removing them.
+ --> lib.cairo:5:14
+        5 => ((20)),
+             ^****^
+
+warning: Plugin diagnostic: unnecessary double parentheses found. Consider removing them.
+ --> lib.cairo:6:14
+        _ => ((30)),
+             ^****^
+
+//! > fixed
+fn main() -> felt252 {
+    let x = 5;
+    match x {
+        1 => 10,
+        5 => 20,
+        _ => 30,
+    }
+}
+
+//! > ==========================================================================
+
+//! > double parens in struct field access
+
+//! > cairo_code
+struct MyStruct {
+    x: felt252,
+    y: felt252,
+}
+
+fn main() -> felt252 {
+    let my_struct = MyStruct { x: 10, y: 20 };
+    return ((my_struct.y));
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: unnecessary double parentheses found. Consider removing them.
+ --> lib.cairo:8:12
+    return ((my_struct.y));
+           ^*************^
+
+//! > fixed
+struct MyStruct {
+    x: felt252,
+    y: felt252,
+}
+
+fn main() -> felt252 {
+    let my_struct = MyStruct { x: 10, y: 20 };
+    return my_struct.y;
+}
+
+//! > ==========================================================================
+
+//! > double parens with function call
+
+//! > cairo_code
+fn foo(x: felt252) -> felt252 {
+    x * 2
+}
+
+fn main() -> felt252 {
+    ((foo(10)))
+}
+
+//! > diagnostics
+warning: Plugin diagnostic: unnecessary double parentheses found. Consider removing them.
+ --> lib.cairo:6:5
+    ((foo(10)))
+    ^*********^
+
+//! > fixed
+fn foo(x: felt252) -> felt252 {
+    x * 2
+}
+
+fn main() -> felt252 {
+    foo(10)}
```

Comment:
This happens because the node text is `((foo(10)))\n` but `\n` is included in the trivia of the outter `(...)`. To fix that we need to extract the leading and trailing trivia of the node and then write it back before and after the inner node.